README.md

```diff
@@ -7,624 +7,796 @@
+> syntax.
 
-To start implementing routes, we first need to import `BrowserRouter` and
-`Route` from `react-router-dom`:
+To start implementing routes, we first need to import `createBrowserRouter` and
+`RouterProvider` from `react-router-dom`:
 
 ```jsx
-// .src/index.js
-
+// index.js
 import React from "react";
-import ReactDOM from "react-dom";
+import ReactDOM from "react-dom/client";
 // Step 1. Import react-router functions
-import { BrowserRouter, Route } from "react-router-dom";
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import Home from "./pages/Home";
 
-function Home() {
-  return (
-    <div>
-      <h1>Home!</h1>
-    </div>
-  );
-}
+const root = ReactDOM.createRoot(document.getElementById("root"));
+root.render(<Home />);
+```
 
-// Step 2. Use <Route> components to define client-side routes in our app
-function App() {
-  return (
-    <Route path="/">
-      <Home />
-    </Route>
-  );
-}
+`createBrowserRouter` is used to create the router for our application. We'll
+pass it an array of route objects as its argument. Each route object will have a
+routing path and a corresponding element to be rendered on that path.
 
-// Step 3. Use <BrowserRouter> component to wrap the entire application and provide React Router context features
-ReactDOM.render(
-  <BrowserRouter>
-    <App />
-  </BrowserRouter>,
-  document.getElementById("root")
-);
+```jsx
+const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <Home />
+  }
+]);
 ```
 
-In the code above, there are two components that we are importing from **React
-Router**. We use them in turn:
+The `RouterProvider` provides the router created by `createBrowserRouter` to our
+application, so it can use React-Router's client-side routing.
 
-1. The `BrowserRouter` component is the base for our application's routing. It
-   is where we declare how **React Router** will be used. Notice that the
-   `BrowserRouter` component is wrapped around our entire application. This lets
-   us use the `Route` component and other React Router components anywhere in
-   our app.
-2. The `Route` component is in charge of saying: "when the URL matches this
-   specified `path`, render this child component." This handles the conditional
-   rendering based on the URL that we described earlier. The `Route` component
-   has one prop in our example: `path`.
+```jsx
+const root = ReactDOM.createRoot(document.getElementById("root"));
+root.render(<RouterProvider router={router} />);
+```
 
-Let's try it. Copy the above code into `src/index.js` and run `npm start` to
+Let's try it! Copy the code below into `src/index.js` and run `npm start` to
 boot up the application. Once it is running, point your URL to
-`http://localhost:3000/`. What you'll notice is that when you type in the URL it
-will render `Home!`.
+`http://localhost:3000/`. We should still see the home page, but now it's being
```

Comment:
```suggestion
`http://localhost:3000/`. We should see the home page being
```

Reply:
I added instructions to run `npm start` earlier in the reading, so this should no longer be necessary.